core/src/main/java/com/opensymphony/xwork2/ognl/SecurityMemberAccess.java

```diff
@@ -95,6 +96,12 @@ public boolean isAccessible(Map context, Object target, Member member, String pr
         LOG.debug("Checking access for [target: {}, member: {}, property: {}]", target, member, propertyName);
 
         final int memberModifiers = member.getModifiers();
+        final Class<?> memberClass = member.getDeclaringClass();
+        // target can be null in case of accessing static fields, since OGNL 3.2.8
+        final Class<?> targetClass = Modifier.isStatic(memberModifiers) ? memberClass : target.getClass();
+        if (!memberClass.isAssignableFrom(targetClass)) {
```

Comment:
Added some validation here and in `#isPackageExcluded` to ensure unit tests made sense